src/extension.ts

```diff
@@ -1732,8 +1744,40 @@ async function deleteContextKubernetes(explorerNode: explorer.KubernetesObject)
 }
 
 async function useNamespaceKubernetes(explorerNode: explorer.KubernetesObject) {
-    if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
-        refreshExplorer();
+    if (explorerNode) {
+        if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
+            refreshExplorer();
+        }
+    } else {
+        promptKindName([kuberesources.allKinds.namespace], undefined,
+            {
+                prompt: 'What namespace do you want to use?',
+                placeHolder: 'Enter the namespace to switch to or press enter to select from available list'
+            },
+            (resource) => {
+                if (resource) {
+                    const currentNS = kubectlUtils.currentNamespace(kubectl);
+                    currentNS.then((ns) => {
+                        let toSwitchNamespace = resource;
+                        // resource will be of format <kind>/<name>, when picked up from the quickpick
+                        if (toSwitchNamespace.lastIndexOf('/') != -1) {
+                            toSwitchNamespace = toSwitchNamespace.substring(toSwitchNamespace.lastIndexOf('/') + 1);
+                        }
+                        // Switch if an only if the currentNS and toSwitchNamespace are different
+                        if (toSwitchNamespace && ns !== toSwitchNamespace) {
+                            const promiseSwitchNS = kubectlUtils.switchNamespace(kubectl, toSwitchNamespace);
```

Comment:
Prefer using `await` here, e.g.:

```
try {
  if (await kubectlUtils.switchNamespace(...)) {
    refreshExplorer();
  }
} catch (err) {
  // ...
}
```

Reply:
learnt how to use async for arrow functions, added it :)